src/main/java/com/volasoftware/tinder/utility/FriendLinker.java

```diff
@@ -24,6 +25,13 @@ public static void linkRandomFriendsForRequestedUser(User requestedUser, List<Us
 
             Set<User> friendsToAdd = generateUniqueRandomFriends(requestedUser, botUsers, randomNumberOfFriends);
             requestedUser.getFriends().addAll(friendsToAdd);
+
+            for (User friendToAdd : friendsToAdd) {
+                if (friendToAdd.getFriends() == null) {
+                    friendToAdd.setFriends(new HashSet<>());
```

Comment:
To prevent every time to set empty Set, try to add a default value (new HashSet<>()) to friends in the User model